frontend/catalyst/compiler.py

```diff
@@ -303,8 +304,11 @@ class Enzyme(PassPipeline):
 
     _executable = get_executable_path("llvm", "opt")
     enzyme_path = get_lib_path("enzyme", "ENZYME_LIB_DIR")
+    apple_ext = "dylib"
+    linux_ext = "so"
+    ext = linux_ext if platform.system() == "Linux" else apple_ext
```

Comment:
@erick-xanadu just wondering if none of the following approaches worked here. As always, avoid hardcoding strings when it's possible.

``` python
import sysconfig
sysconfig.get_config_var('SHLIB_SUFFIX')
```
or, 
``` python
import distutils.ccompiler
tutils.ccompiler.new_compiler().shared_lib_extension
```

Reply:
Ah, that's great! I wasn't aware of these options. Definitely prefer the first one over the second one! Will change. Thanks!

Reply:
Mmm... both of these options still return `.so`. In the past I tried removing the configuration from Enzyme that outputs this [extension](https://github.com/EnzymeAD/Enzyme/blob/main/enzyme/CMakeLists.txt#L38-L40), but I am not sure it can be overridden from the command line (as we invoke Enzyme from a Makefile). Do you see an alternative here?

Reply:
Hmm.. interesting! :thinking: